### DISCUSSION TITLE
Trying to run lib p2p example but it gives error on PUT

### DISCUSSION THREAD

**Discussion by NoorahSmith** [libp2p/rust-libp2p#4329] [Q&A]
Trying to run lib p2p example but it gives error on PUT. 
Running three instances of the same code
Failed to put record: QuorumFailed { key: Key(b"key"), success: [], quorum: 1 }
![image](https://github.com/libp2p/rust-libp2p/assets/136467640/cabc05c0-97e3-4d60-8982-636a33215758)

**Comment by mxinden** [comment]
My first intuition would be that the node is not able to connect to the other nodes. Running with `RUST_LOG=debug` will likely tell you more.

**Reply by NoorahSmith** [reply]
Still riddles . Can you point me at the right direction 
![image](https://github.com/libp2p/rust-libp2p/assets/136467640/c14415f0-b83f-4788-acb8-0adb702a50ae)

**Reply by NoorahSmith** [reply]
Do i need to modify some firewall rules / IP tables  for these to communicate ? IMHO they are listening on localhost so they should communicate without problem . 
the other party gives error as no peers found 
Failed to get record: NotFound { key: Key(b"key1"), closest_peers: [] }

**Reply by mxinden** [reply]
> If your local network supports mDNS, the instances will automatically connect.

Seems like the two nodes don't connect, otherwise you would get a connection established event.

Other examples, e.g. the file-sharing example, don't depend on mdns. Do those work for you?

**Comment by NoorahSmith** [comment]
the local network supports mdns but for some reason , its not connecting in debian or arch . Connecting sometime  in ubuntu server which has different configuration of /etc/nsswith.conf and /etc/avahi/avahi-daemon.conf . 
Will try to copy both configurations and try to make it work on my own systems.

**Comment by NoorahSmith** [comment]
Can any one help me out . The behavior is really weird. Installed Ubuntu 22 LTS. The code successfully put a keyword and then in the same terminal it started giving me error . Is it related to this last commit https://github.com/libp2p/rust-libp2p/commit/cbdbaa836e1159fb4e8b20e76c32f1af5ec66926
![image](https://github.com/libp2p/rust-libp2p/assets/136467640/29f666f5-11d4-4feb-9c87-4bec9ded110b)

**Reply by thomaseizinger** [reply]
Without logs, it is hard to tell whether the nodes are connected at the time of you typing `PUT apple 123`.
